draft-ietf-jsonpath-base.md

```diff
@@ -1577,27 +1660,26 @@ expression, which is in {{-iregexp}} form.
 $[?search(@.author, "[BR]ob")]
 ~~~
 
-Its first argument is an optional string that is searched for at least one
-substring that matches the iregexp contained in the string
-that is the second argument.
-The result is `true` if such a substring exists, `false` otherwise.
-
-The result is `Nothing` if the first argument is not a string or
-the second argument is not a string conforming to {{-iregexp}}.
+Its arguments are instances of ValueType.
+If the first argument is not a string or the second argument is not a
+string conforming to {{-iregexp}}, the result is `LogicalFalse`. <!-- XXX -->
+Otherwise, the string that is the first argument is searched for at
+least one substring that matches the iregexp contained in the string
+that is the second argument; the result is `LogicalTrue` if
 
 
 ### Examples
 {: unnumbered}
 
 | Query | Comment |
 | :---: | ------- |
-| `$[?length(@) < 3]` | Valid typing |
-| `$[?length(@.*) < 3]` | Invalid typing since `@.*` is a non-singular path |
-| `$[?count(@.*) == 1]` | Valid typing |
-| `$[?count(1) == 1]` | Invalid typing since `1` is not a path  |
-| `$[?count(foo(@.*)) == 1]` | Valid typing, where `foo` is a function extension with argument of type `OptionalNodesType` and result type `OptionalNodesType` |
-| `$[?match(@.timezone, 'Europe/.*')]`         | Valid typing |
-| `$[?match(@.timezone, 'Europe/.*') == true]` | Valid typing |
+| `$[?length(@) < 3]` | well-typed |
+| `$[?length(@.*) < 3]` | not well-typed since `@.*` is a non-singular path |
+| `$[?count(@.*) == 1]` | well-typed |
+| `$[?count(1) == 1]` | not well-typed since `1` is not a path  |
+| `$[?count(foo(@.*)) == 1]` | well-typed, where `foo` is a function extension with argument of type `NodesType` and result type `NodesType` |
+| `$[?match(@.timezone, 'Europe/.*')]`         | well-typed |
+| `$[?match(@.timezone, 'Europe/.*') == true]` | not well-typed as JSONPath logicals and JSON booleans do not mix |
```

Comment:
```suggestion
| `$[?match(@.timezone, 'Europe/.*') == true]` | invalid as JSONPath logicals are not well-typed in comparisons |
```

Reply:
I tried to be a bit informal in this example...
(And I certainly don't want to regress from "well-typed" to "invalid" (which has been rooted out, not the least as it can be considered a non-word these days.)

Reply:
```suggestion
| `$[?match(@.timezone, 'Europe/.*') == true]` | not well-typed as JSONPath logicals may not be used in comparisons |
```